js/components/parsers/urlModel_directive.js

```diff
@@ -0,0 +1,34 @@
+angular.module('contactsApp')
+.directive('urlModel', function() {
+	return{
+		restrict: 'A',
+		require: 'ngModel',
+		link: function(scope, element, attr, ngModel) {
+			ngModel.$formatters.push(function(value) {
+				if (typeof value !== 'string') {
```

Comment:
@Henni I get an issue here. All tests using the formater assignment are null at this point! :/
```
$scope.model.url = 'formatters.nextcloud.com';
expect(form.url.$modelValue).to.equal('https://formatters.nextcloud.com');
```
https://github.com/nextcloud/contacts/pull/198/files#diff-19123d57bdd04dbac14aca8a241ae1d2R29

Reply:
Adding `$scope.$digest(); after [this line](https://github.com/nextcloud/contacts/pull/198/files#diff-19123d57bdd04dbac14aca8a241ae1d2R29) fixes it for me.

Reply:
@Henni Yeah, I knew it was going to be something like that  --"